### ISSUE TITLE
Elevator needs some breathing room on the sides

### ISSUE THREAD

**Issue by johnkslg** [StrangeLoopGames/EcoIssues#16425]
![animx22](https://user-images.githubusercontent.com/3536496/82508083-d2d42580-9ab8-11ea-8bdb-75d61b144687.gif)
You get some penetrations when its right flush against the wall

**Comment by johnkslg** [comment]
Push if desired.

**Comment by Conchipedia** [comment]
keep your arms inside the vehicle at all times 😛

**Comment by keegan-o** [comment]
I wonder if there is a way for the elevator to count the blocks it is going through as 'solid', that way the rock builders would not have these decorations that clip-- the decorations only spawn when the builder detects that the block there is empty.

**Comment by keegan-o** [comment]
Yeah... I pulled in the sides and it still clips, would need to be small enough that you could no longer fit a wooden cart in with you, which would be bad to reduce the capability of the elevator like that.
![image](https://user-images.githubusercontent.com/14983165/83190524-658f4880-a0e7-11ea-8f41-9547b06d8658.png)

**Comment by keegan-o** [comment]
Going to push this till we implement the larger elevator, then we can size this one down to be just for one or two people. Or, the solution I mentioned above, but that could be difficult or impossible. (would probably have to mark every space below the elevator's cabin as 'solid' somehow.

**Comment by MilenkoTunjic** [comment]
can you just increase the required space? if it is 2x3, make required hole meter bigger?

**Comment by Conchipedia** [comment]
but wouldn't that make the placing of the model harder? as the placement is only 1 block wider each side than the hole below

**Comment by SlayksWood** [comment]
@keegan-o @johnkslg @WeaselDog 
We added a new elevator, but Industrial Elevator have this problem too.
I think that when placing the elevator under hole, it just has to level the wall.

**Comment by keegan-o** [comment]
![image.png](https://images.zenhubusercontent.com/5ee003d5cfaa7fb4ca69f828/6cde515b-31f4-4f22-bc7a-a366e3a5f479)

Updated all rock builders decorations to not appear when there is a WorldObject block in the space they would take up. This helps with other objects like lights mounted to natural rock, but doesn't fix the elevator problem yet.

This should allow us to hopefully find a way for the elevator object to make blocks where its cabin can transit to be considered occupied by WorldObject blocks. Basically, we want a dynamic world object occupancy for the elevator that updates based on how low the cabin can go, and has the world object update to occupy the space down to that level. This will mean players must block the end of the elevator with blocks, not world objects, but this was essentially the case already and allowing world objects to overlap with the shaft is likely keeping the door open for more problems.
